src/state/modelStore.ts

```diff
@@ -0,0 +1,74 @@
+import { create } from 'zustand'
+import { devtools, persist } from 'zustand/middleware'
+import { upsert_user } from '@app/explore/upsert_user'
+import { Model, TrainingRequest } from '@prisma/client'
+
+type IModel = Omit<Omit<Model, 'createdAt'>, 'updatedAt'>
+type ITrainRequest = Omit<Omit<TrainingRequest, 'createdAt'>, 'updatedAt'>
+
+type Store = {
+  model: IModel | null
```

Comment:
```
type Store = {
  user: IUser | null
  init: (user: IUser) => void
  setUser: (user: Partial<IUser>) => void
}

```

refer the store above its maintaining user state and keeping it in sync with "user" collection in DB

Similarily for every collection in mongoDB, there should be a store. 

It should be like: the stores are our primariy DB and the changes are replicated to mongoDB

Reply:
so there needs to be separate **trainRequest** Store

Reply:
for ModelStore: on init: it should get all models of userId from DB. 

Don't use :   models: IModel[] | null

It becomes harder to access. Use an object : {modelID: model}

Good job on creating:   model: IModel | null to maintain the state of DB-unsynced model. It does raise concerns if the user opens different tabs and tries to create models in new tabs.

Reply:
but that's alright, we can restrict user-usage to just one-tab I guess?

Reply:
> for ModelStore: on init: it should get all models of userId from DB.
> 
> Don't use : models: IModel[] | null
> 
> It becomes harder to access. Use an object : {modelID: model}
> 
> Good job on creating: model: IModel | null to maintain the state of DB-unsynced model. It does raise concerns if the user opens different tabs and tries to create models in new tabs.

I do not understand this object section

Reply:
Ok lets get on the call